### PR TITLE
Use cluster metadata for index list

### PR DESCRIPTION
**Pull request by rzezeski**
Move the index list out of the ring and into cluster metadata. This is
beneficial for several reasons.
- As more data is stored in the ring it becomes larger slowing down
  gossip.
- The larger the ring the more data that must be copied from ETS when
  reading a ring that hasn't settled to mochiglobal.
- No longer need to listen to ring events which can become numerous
  during ownership change and cause listeners to back up on messages.
- No need to send all create messages through the claimant. This had to
  be done when using the ring because the list was stored under one key
  in the ring metadata. Each key handles conflict resolution via a
  physical timestamp which has 1 second resolution. Thus if you update
  same key in same second on disjoint nodes you get data loss. This is
  not an issue with cluster metadata becuase it uses a key _per_ index.

### CODE REVIEW

**Comment by rzezeski** [comment]
Eunit and riak tests passed and dialyzer showed no new errors.

```
Test Results:
yz_wm_extract_test-bitcask   : pass
yz_stat_test-bitcask         : pass
yz_solr_start_timeout-bitcask: pass
yz_siblings-bitcask          : pass
yz_security-bitcask          : pass
yz_schema_admin-bitcask      : pass
yz_rs_migration_test-bitcask : pass
yz_pb-bitcask                : pass
yz_monitor_solr-bitcask      : pass
yz_mapreduce-bitcask         : pass
yz_languages-bitcask         : pass
yz_index_admin-bitcask       : pass
yz_fallback-bitcask          : pass
yz_errors-bitcask            : pass
yz_dt_test-bitcask           : pass
yokozuna_essential-bitcask   : pass
aae_test-bitcask             : pass
---------------------------------------------
0 Tests Failed
17 Tests Passed
That's 100.0% for those keeping score
```

**Comment by rzezeski** [comment]
This is the 2nd and final PR for #168.

File: src/yz_events.erl [hunk 1]
```diff
@@ -96,53 +100,64 @@ code_change(_, S, _) ->
...
-    true = ets:delete(?YZ_EVENTS_TAB),
-    ok.
+%% @private
+%%
+%% @doc Get the number of ticks after which a full check is
+%% performed. If the tick interval is 1 second and this function
+%% returns 60 then a full check will be performed every 60 seconds.
+-spec get_full_check_after() -> non_neg_integer().
+get_full_check_after() ->
+    NumTicks = app_helper:get_env(?YZ_APP_NAME, events_full_check_after, 60),
```
Comment (coderoshi) [inline src/yz_events.erl]: define() magic number?

**Comment by coderoshi** [comment]
```
17 Tests Passed
That's 100.0% for those keeping score
```

Dialyzer is also good.

`yz_misc` claimant code (`get_claimant`, `is_claimant`) don't seem to be used anymore. Delete?

Also +1

**Comment by rzezeski** [comment]
Made the updates and ran tests again just for good measure:

```
Test Results:
yz_wm_extract_test-bitcask   : pass
yz_stat_test-bitcask         : pass
yz_solr_start_timeout-bitcask: pass
yz_siblings-bitcask          : pass
yz_security-bitcask          : pass
yz_schema_admin-bitcask      : pass
yz_rs_migration_test-bitcask : pass
yz_pb-bitcask                : pass
yz_monitor_solr-bitcask      : pass
yz_mapreduce-bitcask         : pass
yz_languages-bitcask         : pass
yz_index_admin-bitcask       : pass
yz_fallback-bitcask          : pass
yz_errors-bitcask            : pass
yz_dt_test-bitcask           : pass
yokozuna_essential-bitcask   : pass
aae_test-bitcask             : pass
---------------------------------------------
0 Tests Failed
17 Tests Passed
That's 100.0% for those keeping score
```

**Pull request merged**